### ISSUE TITLE
Ordering by DateTime

### ISSUE THREAD

**Issue by marcokuoni** [paysera/lib-pagination#3]
Is it possible that i cant order by a DateTime field like CreatedAt? I get all the time the error can't convert to string.

**Comment by vhenzl** [comment]
@lemonbrain-mk, I have run into the same problem and the solution that works for me is to convert `DateTime` to string manually in _accessor closure_:

```php
$configuredQuery = (new ConfiguredQuery($qb))
  ->addOrderingConfiguration(
    'CREATED_AT',
    (new OrderingConfiguration('i.createdAt'))->setAccessorClosure(fn(Incident $x) => $x->getCreatedAt()->format('Y-m-d H:i:s')),
  );
```

Since the string is later used directly in SQL queries, it has to be in the format suitable for your database (`format('Y-m-d H:i:s')` will work for MySQL). A more resilient solution would be to delegate the conversion to the `convertToDatabaseValue()` method of the corresponding Doctrine type (e.g. `Doctrine\DBAL\Types\DateTimeImmutableType`).

Another solution would be to implement your own _cursor builder_ and handle cursor (de)serialisation there. (I haven't tried this approach.)

**Comment by mariusbalcytis** [comment]
Released fix with 1.1.0.

Theoretically – yes, we'd need to "consult" Connection to give representation of DateTime.
As this would need backwards incompatible changes and additional requirements, I chose a lot simpler solution, which will probably work in all situations this library would be used with.

Thanks for noticing and reporting.
